### PR TITLE
Add org.darkdimension.rue

### PR DESCRIPTION
**Pull request by tomtsagk**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

For the permissions, this is a 3D game that needs a graphical environment and OpenGL to run. From what I've seen, other games use similar permissions. Happy to provide more details if they are considered non-standard.

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: org.darkdimension.rue.json [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+			"buildsystem": "simple",
+			"build-commands": [
+				"make prefix=${FLATPAK_DEST} avdlargs=\"-L ${FLATPAK_DEST}/lib\"",
+				"make prefix=${FLATPAK_DEST} install"
+			],
+			"sources": [
+				{
+					"type": "archive",
+					"url": "https://github.com/tomtsagk/rue/archive/refs/tags/v0.0.7.tar.gz",
+					"md5": "6843e1f0f42a1d79e58a82bf55c2fe9c"
```
Comment (hfiguiere) [inline org.darkdimension.rue.json:42]: please use sha256 sums. md5 is deprecated.

File: org.darkdimension.rue.json [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+			"buildsystem": "simple",
+			"build-commands": [
+				"make prefix=",
+				"make DESTDIR=${FLATPAK_DEST} prefix= install"
+			],
+			"sources": [
+				{
+					"type": "archive",
+					"url": "https://github.com/tomtsagk/avdl/archive/refs/tags/v0.1.3.tar.gz",
+					"md5": "3d4fea5a6f4b1e1856f2aa4b82b55250"
```
Comment (hfiguiere) [inline org.darkdimension.rue.json:21]: please use sha256 instead

File: org.darkdimension.rue.json [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+	"sdk": "org.freedesktop.Sdk",
+	"command": "rue",
+	"modules": [
+		"shared-modules/glew/glew.json",
+		"shared-modules/glu/glu-9.json",
+		{
+			"name": "avdl",
+			"buildsystem": "simple",
+			"build-commands": [
+				"make prefix=",
```
Comment (hfiguiere) [inline org.darkdimension.rue.json:14]: shouldn't prefix be `${FLATPAK_DEST}` ?

**Comment by hfiguiere** [comment]
bot, build org.darkdimension.rue

**Review by hfiguiere — CHANGES_REQUESTED** [review]

**Comment by hfiguiere** [comment]
> For the permissions, this is a 3D game that needs a graphical environment and OpenGL to run. From what I've seen, other games use similar permissions. Happy to provide more details if they are considered non-standard.

For 3D you only need `--device=dri`. But then, gamepads.

File: org.darkdimension.rue.json [hunk 4]
```diff
@@ -0,0 +1,53 @@
+{
+	"id": "org.darkdimension.rue",
+	"runtime": "org.freedesktop.Platform",
+	"runtime-version": "20.08",
+	"sdk": "org.freedesktop.Sdk",
+	"command": "rue",
+	"modules": [
+		"shared-modules/glew/glew.json",
```
Comment (hfiguiere) [inline org.darkdimension.rue.json:8]: you need to add the `shared-modules` git submodule for this to work.

File: org.darkdimension.rue.json (hunk 1 above)
Comment (tomtsagk) [reply org.darkdimension.rue.json]: I've updated the `md5` field to use the `sha256` sum instead.

File: org.darkdimension.rue.json (hunk 2 above)
Comment (tomtsagk) [reply org.darkdimension.rue.json]: I've updated the `md5` field to use the `sha256` sum instead.

File: org.darkdimension.rue.json (hunk 3 above)
Comment (tomtsagk) [reply org.darkdimension.rue.json]: That is correct, I missed it and passed it during the installation step only. It is fixed now to be part of `prefix=`

File: org.darkdimension.rue.json (hunk 4 above)
Comment (tomtsagk) [reply org.darkdimension.rue.json:8]: Thanks, I've added `shared-modules` as a submodule as part of this pull request.

**Comment by tomtsagk** [comment]
> > For the permissions, this is a 3D game that needs a graphical environment and OpenGL to run. From what I've seen, other games use similar permissions. Happy to provide more details if they are considered non-standard.
> 
> For 3D you only need `--device=dri`. But then, gamepads.

I've changed this to use `--device=dri` only, and it all seemed to work locally. I haven't added gamepad functionality yet, so this works for this project. I'll revisit it once gamepad support is added in a future version.

**Comment by hfiguiere** [comment]
bot, build org.darkdimension.rue

**Comment by hfiguiere** [comment]
appdata is missing screenshots.

**Comment by tomtsagk** [comment]
> appdata is missing screenshots.

Apologies, I must have missed that they are required. I've updated the project to include screenshots in the appdata, and updated this PR to the new version `0.0.8` that includes them.

**Comment by nedrichards** [comment]
bot, build org.darkdimension.rue

**Pull request closed without merging**